src/core/libraries/rtc/rtc.cpp

```diff
@@ -182,69 +423,277 @@ int PS4_SYSV_ABI sceRtcSetCurrentTick() {
     return ORBIS_OK;
 }
 
-int PS4_SYSV_ABI sceRtcSetDosTime() {
-    LOG_ERROR(Lib_Rtc, "(STUBBED) called");
-    return ORBIS_OK;
+int PS4_SYSV_ABI sceRtcSetDosTime(OrbisRtcDateTime* pTime, u32 dosTime) {
+    LOG_TRACE(Lib_Rtc, "called");
+
+    if (pTime == nullptr)
+        return ORBIS_RTC_ERROR_INVALID_POINTER;
+
+    pTime->microsecond = 0;
+    pTime->second = (dosTime << 1) & 0x3e;
+    pTime->minute = (dosTime >> 5) & 0x3f;
+    pTime->hour = ((dosTime & 0xf800) >> 0xb);
+
+    int days = (dosTime >> 0x10);
+
+    pTime->day = days & 0x1f;
+    pTime->month = (days >> 5) & 0xf;
+    pTime->year = (days >> 9) + 0x7bc;
+
+    return 0;
 }
 
-int PS4_SYSV_ABI sceRtcSetTick() {
-    LOG_ERROR(Lib_Rtc, "(STUBBED) called");
-    return ORBIS_OK;
+int PS4_SYSV_ABI sceRtcSetTick(OrbisRtcDateTime* pTime, OrbisRtcTick* pTick) {
+    LOG_TRACE(Lib_Rtc, "called");
+
+    if (pTime == nullptr || pTick == nullptr)
+        return ORBIS_RTC_ERROR_INVALID_POINTER;
+
+    u32 ly, ld, lm, j;
+    u64 days, msec;
+
+    days = pTick->tick / 86400000000;
+    msec = pTick->tick % 86400000000;
+
+    days += 307;
+
+    j = (days << 2) - 1;
+    ly = j / 146097;
+
+    j -= (146097 * ly);
+    ld = j >> 2;
+
+    j = ((ld << 2) + 3) / 1461;
+    ld = (((ld << 2) + 7) - 1461 * j) >> 2;
+
+    lm = (5 * ld - 3) / 153;
+    ld = (5 * ld + 2 - 153 * lm) / 5;
+    ly = 100 * ly + j;
+
+    if (lm < 10) {
+        lm += 3;
+    } else {
+        lm -= 9;
+        ly++;
+    }
+
+    pTime->year = ly;
+    pTime->month = lm;
+    pTime->day = ld;
+
+    pTime->hour = msec / 3600000000;
+    msec %= 3600000000;
+    pTime->minute = msec / 60000000;
+    msec %= 60000000;
+    pTime->second = msec / 1000000;
+    msec %= 1000000;
+    pTime->microsecond = msec;
+
+    return 0;
 }
 
-int PS4_SYSV_ABI sceRtcSetTime_t() {
-    LOG_ERROR(Lib_Rtc, "(STUBBED) called");
+int PS4_SYSV_ABI sceRtcSetTime_t(OrbisRtcDateTime* pTime, u64 iTime) {
+    LOG_TRACE(Lib_Rtc, "called");
+
+    if (pTime == nullptr)
+        return ORBIS_RTC_ERROR_INVALID_POINTER;
+
+    int sdk_version;
+    int sdkResult = Kernel::sceKernelGetCompiledSdkVersion(&sdk_version);
+    if (sdkResult != ORBIS_OK) {
+        sdk_version = 0;
+    }
+
+    OrbisRtcTick newTick;
+    if (sdk_version < 0x3000000) {
+        newTick.tick = (iTime & 0xffffffff) * 1000000;
+    } else {
+        if (iTime < 0) {
+            return ORBIS_RTC_ERROR_INVALID_VALUE;
+        }
+        newTick.tick = iTime * 1000000;
+    }
+
+    newTick.tick += 0xdcbffeff2bc000;
+    sceRtcSetTick(pTime, &newTick);
+
     return ORBIS_OK;
 }
 
-int PS4_SYSV_ABI sceRtcSetWin32FileTime() {
-    LOG_ERROR(Lib_Rtc, "(STUBBED) called");
-    return ORBIS_OK;
+int PS4_SYSV_ABI sceRtcSetWin32FileTime(OrbisRtcDateTime* pTime, u64 win32Time) {
```

Comment:
```suggestion
int PS4_SYSV_ABI sceRtcSetWin32FileTime(OrbisRtcDateTime* pTime, uint64_t ulWin32Time) {
```
Same for `sceRtcGetWin32FileTime`, with `*ulWin32Time`.